javav2/example_code/s3/src/main/java/com/example/s3/express/S3DirectoriesScenario.java

```diff
@@ -0,0 +1,578 @@
+import software.amazon.awssdk.services.ec2.model.CreateVpcEndpointRequest;
+import software.amazon.awssdk.services.ec2.model.CreateVpcEndpointResponse;
+import software.amazon.awssdk.services.ec2.model.CreateVpcRequest;
+import software.amazon.awssdk.services.ec2.model.CreateVpcResponse;
+import software.amazon.awssdk.services.ec2.model.DescribeAvailabilityZonesRequest;
+import software.amazon.awssdk.services.ec2.model.DescribeAvailabilityZonesResponse;
+import software.amazon.awssdk.services.ec2.model.DescribeRouteTablesRequest;
+import software.amazon.awssdk.services.ec2.model.DescribeRouteTablesResponse;
+import software.amazon.awssdk.services.ec2.model.DescribeVpcsRequest;
+import software.amazon.awssdk.services.ec2.model.Ec2Exception;
+import software.amazon.awssdk.services.ec2.model.Filter;
+import software.amazon.awssdk.services.ec2.waiters.Ec2Waiter;
+import software.amazon.awssdk.services.iam.IamAsyncClient;
+import software.amazon.awssdk.services.iam.IamClient;
+import software.amazon.awssdk.services.iam.model.CreateAccessKeyRequest;
+import software.amazon.awssdk.services.iam.model.CreateAccessKeyResponse;
+import software.amazon.awssdk.services.iam.model.IamException;
+import software.amazon.awssdk.services.s3.S3AsyncClient;
+import software.amazon.awssdk.services.s3.model.S3Exception;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Scanner;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+import java.util.stream.IntStream;
+
+public class S3DirectoriesScenario {
+
+    public static final String DASHES = new String(new char[80]).replace("\0", "-");
+
+    private static final Logger logger = LoggerFactory.getLogger(S3DirectoriesScenario.class);
+    static Scanner scanner = new Scanner(System.in);
+
+    private static S3AsyncClient mS3RegularClient;
+    private static S3AsyncClient mS3ExpressClient;
+
+    private static String mdirectoryBucketName;
+    private static String mregularBucketName;
+
+    private static String stackName = "cfn-stack-s3-express-basics--" + UUID.randomUUID();
+
+    private static String regularUser = "";
+    private static String vpcId = "";
+    private static String expressUser = "";
+
+    private static String vpcEndpointId = "";
+
+    private static final S3DirectoriesActions s3DirectoriesActions = new S3DirectoriesActions();
+
+    public static void main(String[] args) {
+        try {
+            s3ExpressScenario();
+        } catch (RuntimeException e) {
+            logger.info(e.getMessage());
+        }
+    }
+
+    // Runs the scenario.
+    private static void s3ExpressScenario() {
+        logger.info(DASHES);
+        logger.info("Welcome to the Amazon S3 Express Basics demo using AWS SDK for Java V2");
+        logger.info("""
+            Let's get started! First, please note that S3 Express One Zone works best when working within the AWS infrastructure,
```

Comment:
I don't think we should suggested running this example on infrastructure unless we are going to provide a runnable jar file for users that they can put on an EC2 instance (or however else they want to run it on AWS infrastructure). If we do provide a runnable jar, give them instrux on where to find it and how to run it.

Reply:
Lets look at changing the text. I agree - its a bit confusing.